From 191142172d118ca289f08cf93bffe8c0ce1426c4 Mon Sep 17 00:00:00 2001
From: Santiago Seifert <aquilescanta@google.com>
Date: Thu, 19 May 2022 15:29:26 +0000
Subject: [PATCH] Avoid read out of bounds

Bug: 230493653
Change-Id: Ieca5a5390d3cf73fff6aa552d065363d84e1ccc2
Merged-In: Ieca5a5390d3cf73fff6aa552d065363d84e1ccc2
Test: See bug for PoC.
(cherry picked from commit 306aad773337f228bffcf5bf07a3e6663226f42c)
(cherry picked from commit 9d33304ec75b366ed9750e7bde6f96f8c704e1c8)
Merged-In: Ieca5a5390d3cf73fff6aa552d065363d84e1ccc2
---
 media/extractors/mp4/MPEG4Extractor.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/media/extractors/mp4/MPEG4Extractor.cpp b/media/extractors/mp4/MPEG4Extractor.cpp
index f157d359b2..78e3f00b44 100644
--- a/media/extractors/mp4/MPEG4Extractor.cpp
+++ b/media/extractors/mp4/MPEG4Extractor.cpp
@@ -4573,7 +4573,7 @@ status_t MPEG4Extractor::updateAudioTrackInfoFromESDS_MPEG4Audio(
         if (len2 == 0) {
             return ERROR_MALFORMED;
         }
-        if (offset >= csd_size || csd[offset] != 0x01) {
+        if (offset + len1 > csd_size || csd[offset] != 0x01) {
             return ERROR_MALFORMED;
         }
         // formerly kKeyVorbisInfo
-- 
2.37.0.rc0.161.g10f37bed90-goog

